README.md

```diff
@@ -1,5 +1,87 @@
 
-# Contributing
+# Microsoft Authentication Extensions for Python
+
+The Microsoft Authentication Extensions for Python offers secure mechanisms for public client applications to perform cross-platform token cache serialization and persistence. It gives additional support to the [Microsoft Authentication Library for Python (MSAL)](https://github.com/AzureAD/microsoft-authentication-library-for-python). 
+
+MSAL Python supports an in-memory cache by default and provides the [SerializableTokenCache](https://msal-python.readthedocs.io/en/latest/#msal.SerializableTokenCache) to perform cache serialization. You can read more about this in the MSAL Python [documentation](https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-python-token-cache-serialization). Developers are required to implement their own cache persistance across multiple platforms and Microsoft Authentication Extensions makes this simpler. The extensions library supports persistence of the cache to disk in encrypted form where the platform supports this.
+
+This is available for Windows, Mac and Linux. 
+- Windows - [DPAPI](https://docs.microsoft.com/en-us/dotnet/standard/security/how-to-use-data-protection) is used for encryption.
+- MAC - The MAC KeyChain is used and encryption is built in.
+- Linux - [LibSecret](https://wiki.gnome.org/Projects/Libsecret) is used for encryption. Any place where libsecret is not available encryption is not supported.
+
+> Note: This library provides cache persistance support for Public client applications such as Desktop apps only. Web applications are recommended to persist the cache in memory or in Redis. Take a look at this [webapp sample](https://github.com/Azure-Samples/ms-identity-python-webapp).

```

Comment:
```suggestion
> Note: This library provides cache persistance support for Public client applications such as Desktop apps only. Web applications are recommended to persist the cache in session. Take a look at this [webapp sample](https://github.com/Azure-Samples/ms-identity-python-webapp).

```

I think `session` is a more generic yet more precise term here. Our sample is not using cache in memory, and is not demonstrating Redis.